manager/orchestrator/update/updater.go

```diff
@@ -320,7 +321,18 @@ func (u *Updater) worker(ctx context.Context, queue <-chan orchestrator.Slot, up
 			cleanTask   *api.Task
 		)
 		for _, t := range slot {
-			if !u.isTaskDirty(t) {
+			// Don't need a new task if:
+			// 1. Either the task spec didn't change at all, or
+			// 2. The placement constraints alone changed and the node currently assigned
+			// can satisfy the changed constraints.
+			placementSatisfied := u.isTaskPlacementConstraintsOnlyDirty(t) && u.nodeMatches(t)
+			if placementSatisfied || !u.isTaskDirty(t) {
+				if placementSatisfied {
+					if err := u.updateTaskServiceSpec(t); err != nil {
```

Comment:
I think a new task is needed if update fails.

Reply:
Why touch the task spec at all? If the orchestrator knows that the task doesn't need to be updated, it doesn't seem necessary.

Reply:
I am not sure what you mean. This is the case where the placement constraints in the task spec have changed, and this change is made under the assumption that the change needs to be reflected in the task spec.

I don't know if there will be any side effects of not updating the task spec in the existing task.